temporalcli/commands.workflow.go

```diff
@@ -22,8 +27,33 @@ func (*TemporalWorkflowResetBatchCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
-func (*TemporalWorkflowSignalCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowSignalCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	// Get input payloads
+	input, err := c.buildRawInputPayloads()
+	if err != nil {
+		return err
+	}
+
+	// Send signal. We have to use the raw signal service call here because the Go
+	// SDK's signal call doesn't accept multiple arguments.
+	_, err = cl.WorkflowService().SignalWorkflowExecution(cctx, &workflowservice.SignalWorkflowExecutionRequest{
+		Namespace:         c.Parent.Namespace,
+		WorkflowExecution: &common.WorkflowExecution{WorkflowId: c.WorkflowId, RunId: c.RunId},
+		SignalName:        c.Name,
+		Input:             input,
+		Identity:          clientIdentity(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed signalling workflow: %w", err)
+	}
+	cctx.Printer.Println("Signal workflow succeeded")
```

Comment:
Up for discussion on whether this is a log statement or a normal line. If `--output json` is specified, there will be no output regardless.

Reply:
I'm not personally expecting a non-blocking request-response command such as `signal` to have a concept of "logging" -- I'm just expecting stdout and stderr.

Reply:
:+1: This matches existing CLI where, in the absence of more detail, it just prints a line of success.